src/horus_api.c

```diff
@@ -68,16 +68,44 @@ int8_t uw_horus_rtty[] = {
   0,0,1,0,0,1,0,1,1,0
 };
 
-/* Unique word for Horus Binary */
+/* Unique word for Horus Legacy Binary */
 
-int8_t uw_horus_binary[] = {
+int8_t uw_horus_legacy_binary[] = {
     0,0,1,0,0,1,0,0,
     0,0,1,0,0,1,0,0 
 };
 
-struct horus *horus_open (int mode, int Rs) {
-    int i;
-    assert((mode == HORUS_MODE_RTTY) || (mode == HORUS_MODE_BINARY));
+
+/* Unique word for Horus Binary 128/256 bit modes (Last row in the 32x32 Hadamard matrix) */
+
+int8_t uw_horus_binary_new[] = {
+    1, 0, 0, 1, 0, 1, 1, 0,  // 0x96
+    0, 1, 1, 0, 1, 0, 0, 1,  // 0x69
+    0, 1, 1, 0, 1, 0, 0, 1,  // 0x69
+    1, 0, 0, 1, 0, 1, 1, 0   // 0x96
+};
+
+
+struct horus *horus_open (int mode) {
+    assert((mode == HORUS_MODE_RTTY) || (mode == HORUS_MODE_BINARY) || (mode == HORUS_MODE_BINARY_256BIT) || (mode == HORUS_MODE_BINARY_128BIT));
+
+    if (mode == HORUS_MODE_RTTY){
+        // RTTY Mode defaults - 100 baud, no assumptions about tone spacing.
+        return horus_open_advanced(HORUS_MODE_RTTY, HORUS_RTTY_DEFAULT_BAUD, -1);
+    } else {
+        // Placeholder until we have more definition of the new modes.
+
+        // Legacy Horus Binary Mode defaults - 100 baud, 270 Hz Tone spacing (RS41HUP Transmitter)
+        return horus_open_advanced(HORUS_MODE_BINARY, HORUS_BINARY_DEFAULT_BAUD, HORUS_BINARY_DEFAULT_TONE_SPACING);
```

Comment:
Hmmm, hard coding the protocol defaults to specific Tx hardware?  Maybe this should be an init-time value?

Reply:
Since there's no way of entering this kind of setting into freedv, we need to have default somewhere. This is the point of the simplified horus_open function - it uses these defaults.
horus_open_advanced provides the option to either use the default (by using -1 for Rs or tx_tone_separation), or providing a value.